### PR TITLE
Drop support for checking object with `path` property in function returns by `isGitIgnored`

### PR DESCRIPTION
**Pull request by fisker**
Ref: https://github.com/sindresorhus/globby/pull/207#issuecomment-1014263028

This behavior is never documented.

### CODE REVIEW

**Comment by fisker** [comment]
I found other bugs...

File: index.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ const checkCwdOption = options => {
 	}
 };
 
-const getPathString = p => p.stats instanceof fs.Stats ? p.path : p;
+const getPathString = fastGlobResult => fastGlobResult.path || fastGlobResult;
```
Comment (fisker) [inline index.js:37]: This function mean to handle fastGlobResult, but `p.stats` is not always true, see  [`objectMode` option in `fast-glob`](https://github.com/mrmlnc/fast-glob#objectmode)

File: index.js [hunk 2]
```diff
@@ -152,7 +152,7 @@ export const globby = async (patterns, options = {}) => {
 	const [filter, tasks] = await Promise.all([getFilter(options), getTasks()]);
 	const paths = await Promise.all(tasks.map(task => fastGlob(task.pattern, task.options)));
 
-	return arrayUnion(...paths).filter(path_ => !filter(getPathString(path_)));
```
Comment (fisker) [inline index.js:145]: `filter` already handles `objectMode` result, not needed here, `globbySync` and `globbyStream` also not doing this.

**Comment by fisker** [comment]
This one is ready, but I saw another bug here, the `arrayUnion` use is not correct, when use `fastGlob` with `objectMode`, `arrayUnion` can't work, we should implement a function to unique the fastGlob result with `path`, I'll send another PR.

**Comment by sindresorhus** [comment]
Thanks for catching these bugs. I really appreciate it.

**Pull request merged**